# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.h
# Commit: 9c8b6e35bf3f
# Full Hash: 9c8b6e35bf3f9916e114ced0473be4b778c7bab1
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 1 - CookieStore API - WebIDL, r=smaug,edgul
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215144
# ==============================================================================

diff -r a9b3b52f33a5 -r 9c8b6e35bf3f dom/workers/WorkerScope.h
--- a/dom/workers/WorkerScope.h	Tue Sep 10 16:21:33 2024 +0000
+++ b/dom/workers/WorkerScope.h	Tue Sep 10 16:31:38 2024 +0000
@@ -55,6 +55,7 @@
 class ClientSource;
 class Clients;
 class Console;
+class CookieStore;
 class Crypto;
 class DOMString;
 class DebuggerNotificationManager;
@@ -482,6 +483,8 @@
 
   void EventListenerAdded(nsAtom* aType) override;
 
+  already_AddRefed<mozilla::dom::CookieStore> CookieStore();
+
   IMPL_EVENT_HANDLER(message)
   IMPL_EVENT_HANDLER(messageerror)
 
@@ -491,6 +494,8 @@
   IMPL_EVENT_HANDLER(push)
   IMPL_EVENT_HANDLER(pushsubscriptionchange)
 
+  IMPL_EVENT_HANDLER(cookiechange)
+
  private:
   ~ServiceWorkerGlobalScope();
 
@@ -500,6 +505,7 @@
   const nsString mScope;
   RefPtr<ServiceWorkerRegistration> mRegistration;
   SafeRefPtr<extensions::ExtensionBrowser> mExtensionBrowser;
+  RefPtr<mozilla::dom::CookieStore> mCookieStore;
 };
 
 class WorkerDebuggerGlobalScope final : public WorkerGlobalScopeBase {